### ISSUE TITLE
[Bug] Log level not honored

### ISSUE THREAD

**Issue by abh** [strimzi/strimzi-kafka-operator#1918]
**Describe the bug**
Log level set to WARN, but INFO logs still appearing

**To Reproduce**

I configured the topicOperator and zookeeper like this

```
  zookeeper:
    replicas: 3
    storage:
      type: persistent-claim
      size: 5Gi
      deleteClaim: true
    resources:
      requests:
        memory: 100Mi
        cpu: 100m
      limits:
        memory: 1Gi
        cpu: 1
    logging:
      type: inline
      loggers:
        logger.name: "WARN"

  entityOperator:
    topicOperator: {}
      logging:
        type: inline
        loggers:
          logger.name: "WARN"
    userOperator: {}
```

**Expected behavior**
INFO logs not appearing.

**Environment (please complete the following information):**
 - Strimzi version: 0.13
 - Installation method: Yaml files
 - Kubernetes cluster: 1.14
 - Infrastructure: Packet

**YAML files and logs**

```
$ kubectl -n kafka logs kube-logs-entity-operator-94ff88666-8677b topic-operator | tail -5
[2019-08-23 03:38:35,636] INFO  <opicOperator:1184> [oop-thread-0] Success reconciling KafkaTopic kafka/logs-ask
[2019-08-23 03:38:35,730] INFO  <opicOperator:460> [oop-thread-0] 111|periodic kafka __consumer_offsets|51086697: Reconciling topic __consumer_offsets, k8sTopic:nonnull, kafkaTopic:nonnull, privateTopic:nonnull
[2019-08-23 03:38:35,731] INFO  <opicOperator:599> [oop-thread-0] 111|periodic kafka __consumer_offsets|51086697: All three topics are identical
[2019-08-23 03:38:35,731] INFO  <opicOperator:1184> [oop-thread-0] Success reconciling KafkaTopic kafka/consumer-offsets---84e7a678d08f4bd226872e5cdd4eb527fadc1c6a
2019-08-23T03:38:35.733+0000: 568.981: [GC (Allocation Failure) 2019-08-23T03:38:35.733+0000: 568.981: [DefNew: 6319K->139K(6848K), 0.0045608 secs] 17558K->11379K(21732K), 0.0046757 secs] [Times: user=0.01 sys=0.00, real=0.01 secs]
```

```
$ kubectl -n kafka logs kube-logs-zookeeper-0  zookeeper | tail -10
2019-08-23 03:39:30,508 INFO Closed socket connection for client /127.0.0.1:51588 (no session established for client) (org.apache.zookeeper.server.NIOServerCnxn) [Thread-139]
2019-08-23 03:39:36,425 INFO Accepted socket connection from /127.0.0.1:52206 (org.apache.zookeeper.server.NIOServerCnxnFactory) [NIOServerCxn.Factory:0.0.0.0/0.0.0.0:21810]
2019-08-23 03:39:36,426 INFO Processing ruok command from /127.0.0.1:52206 (org.apache.zookeeper.server.NIOServerCnxn) [NIOServerCxn.Factory:0.0.0.0/0.0.0.0:21810]
2019-08-23 03:39:36,427 INFO Closed socket connection for client /127.0.0.1:52206 (no session established for client) (org.apache.zookeeper.server.NIOServerCnxn) [Thread-140]
2019-08-23 03:39:40,525 INFO Accepted socket connection from /127.0.0.1:52636 (org.apache.zookeeper.server.NIOServerCnxnFactory) [NIOServerCxn.Factory:0.0.0.0/0.0.0.0:21810]
2019-08-23 03:39:40,526 INFO Processing ruok command from /127.0.0.1:52636 (org.apache.zookeeper.server.NIOServerCnxn) [NIOServerCxn.Factory:0.0.0.0/0.0.0.0:21810]
2019-08-23 03:39:40,527 INFO Closed socket connection for client /127.0.0.1:52636 (no session established for client) (org.apache.zookeeper.server.NIOServerCnxn) [Thread-141]
2019-08-23 03:39:46,375 INFO Accepted socket connection from /127.0.0.1:53306 (org.apache.zookeeper.server.NIOServerCnxnFactory) [NIOServerCxn.Factory:0.0.0.0/0.0.0.0:21810]
2019-08-23 03:39:46,376 INFO Processing ruok command from /127.0.0.1:53306 (org.apache.zookeeper.server.NIOServerCnxn) [NIOServerCxn.Factory:0.0.0.0/0.0.0.0:21810]
2019-08-23 03:39:46,377 INFO Closed socket connection for client /127.0.0.1:53306 (no session established for client) (org.apache.zookeeper.server.NIOServerCnxn) [Thread-142]
```

```
$ kubectl -n kafka get cm kube-logs-zookeeper-config  -o json | jq -r '.data["log4j.properties"]'
# Do not change this generated file. Logging can be configured in the corresponding kubernetes/openshift resource.
log4j.appender.CONSOLE=org.apache.log4j.ConsoleAppender
log4j.appender.CONSOLE.layout=org.apache.log4j.PatternLayout
log4j.appender.CONSOLE.layout.ConversionPattern=%d{ISO8601} %p %m (%c) [%t]%n
zookeeper.root.logger=INFO
log4j.rootLogger=${zookeeper.root.logger}, CONSOLE
logger.name=WARN
```

**Comment by scholzj** [comment]
The `logger.name` has to be replaced with a name of the actual logger. The name of the loggers for Zookeeper is `zookeeper.root.logger` (https://strimzi.io/docs/latest/full.html#ref-loggers-Kafka-deployment-configuration-kafka) and for the Topic Operator it is `rootLogger.level` (https://strimzi.io/docs/latest/full.html#topic-operator-deployment-configuration-kafka).

So please try to update your resource to look something like this:

```yaml
  zookeeper:
    # ...
    logging:
      type: inline
      loggers:
        zookeeper.root.logger: "WARN"
  entityOperator:
    topicOperator: {}
      logging:
        type: inline
        loggers:
          rootLogger.level: "WARN"
    # ...
```

That should help.

**Comment by scholzj** [comment]
@laidan6000 @PaulRMellor Could you please have a look at the Docs to see whether we can improve something to make this more clear? Thanks.

**Comment by scholzj** [comment]
Fixed in #1922
